applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -0,0 +1,359 @@
+{
+	int err;
+	struct cloud_msg msg = {
+		.qos = CLOUD_QOS_AT_MOST_ONCE,
+		.endpoint.type = CLOUD_EP_STATE,
+		.buf = SERVICE_INFO_GPS,
+		.len = strlen(SERVICE_INFO_GPS)
+	};
+
+	ARG_UNUSED(backend);
+	ARG_UNUSED(user_data);
+
+	switch (evt->type) {
+	case CLOUD_EVT_CONNECTING:
+		LOG_DBG("CLOUD_EVT_CONNECTING");
+		break;
+	case CLOUD_EVT_CONNECTED:
+		LOG_INF("CLOUD_EVT_CONNECTED");
+		break;
+	case CLOUD_EVT_READY:
+		LOG_INF("CLOUD_EVT_READY");
+		/* Update nRF Cloud with GPS service info signifying GPS capabilities. */
+		err = cloud_send(nrf_cloud, &msg);
+		if (err) {
+			LOG_WRN("Failed to send message to cloud, error: %d", err);
+		}
+		/* Start GPS module, which would request back of A-GPS data. */
+		err = gps_start(gps_dev, &gps_cfg);
+		if (err) {
+			LOG_ERR("Failed to start AGPS, error: %d", err);
+		} else {
+			ttft_start = k_uptime_get();
+		}
+		break;
+	case CLOUD_EVT_DISCONNECTED:
+		LOG_INF("CLOUD_EVT_DISCONNECTED");
+		break;
+	case CLOUD_EVT_ERROR:
+		LOG_ERR("CLOUD_EVT_ERROR");
+		break;
+	case CLOUD_EVT_DATA_SENT:
+		LOG_DBG("CLOUD_EVT_DATA_SENT");
+		break;
+	case CLOUD_EVT_DATA_RECEIVED:
+		LOG_INF("CLOUD_EVT_DATA_RECEIVED");
+		err = gps_process_agps_data(evt->data.msg.buf, evt->data.msg.len);
+		if (err) {
+			LOG_INF("Unable to process A-GPS data, error: %d", err);
+		}
+		break;
+	case CLOUD_EVT_PAIR_REQUEST:
+		LOG_DBG("CLOUD_EVT_PAIR_REQUEST");
+		break;
+	case CLOUD_EVT_PAIR_DONE:
+		LOG_DBG("CLOUD_EVT_PAIR_DONE");
+		break;
+	case CLOUD_EVT_FOTA_DONE:
+		LOG_DBG("CLOUD_EVT_FOTA_DONE");
+		break;
+	case CLOUD_EVT_FOTA_ERROR:
+		LOG_ERR("CLOUD_EVT_FOTA_ERROR");
+		break;
+	default:
+		break;
+	}
+}
+#endif /* CONFIG_SLM_AGPS */
+
+/**@brief handle AT#XGPS commands
+ *  AT#XGPS=<op>[,<interval>[,<timeout>]]
+ *  AT#XGPS? READ command not supported
+ *  AT#XGPS=?
+ */
+int handle_at_gps(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t op;
+	uint16_t interval;
+	uint32_t timeout = 0;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		err = at_params_unsigned_short_get(&at_param_list, 1, &op);
+		if (err < 0) {
+			return err;
+		}
+		if (op == GPS_START && gps_dev != NULL) {
+			err = at_params_unsigned_short_get(&at_param_list, 2, &interval);
+			if (err < 0) {
+				return err;
+			}
+			(void)at_params_unsigned_int_get(&at_param_list, 3, &timeout);
+			/* GNSS API spec check */
+			if (interval == 0 || interval == 1 ||
+			   (interval >= 10 && interval <= 1800)) {
+				if (interval == 0) {
+					gps_cfg.nav_mode = GPS_NAV_MODE_SINGLE_FIX;
+				} else if (interval == 1) {
+					gps_cfg.nav_mode = GPS_NAV_MODE_CONTINUOUS;
+				} else {
+					gps_cfg.nav_mode = GPS_NAV_MODE_PERIODIC;
+				}
+				gps_cfg.interval   = interval;
+				gps_cfg.timeout    = timeout;
+
+				err = gps_start(gps_dev, &gps_cfg);
+				if (err) {
+					LOG_ERR("Failed to start GPS, error: %d", err);
+				} else {
+					ttft_start = k_uptime_get();
+				}
+			}
+		} else if (op == GPS_STOP) {
```

Comment:
You have a NULL check for GPS_START and not for GPS_STOP? I'd be consistent here - either return early if GPS device is not initialized, or just ignore it (the gps_start/stop API has NULL pointer checks).

Reply:
Add check at the entry of Set command